src/dataflow/utilities/node.ts

```diff
@@ -267,6 +267,9 @@ export interface NodeChannelInfo {
   units: string;
   plug: number;
   value: number;
+  virtual?: boolean;
+  name: string;
+  method?: (t: number) => number;
```

Comment:
Not sure if it makes much difference one way or the other, but when there are multiple optional properties that depend on each other (if there is `virtual` then we must have `method`) then it can be nice to instead extend the class instead (`VirtualNodeChannelInfo` maybe) and make them both required. That way Typescript can be sure that if one exists then so does the other.

You could then also implement a type guard, so instead of having to check both properties like
```
if (chInfo?.virtual) {
   chInfo.value = chInfo.method ? chInfo.method(time) : 0;
}
```

you could have 

```
function isVirtual(info: NodeChannelInfo | VirtualNodeChannelInfo): info is VirtualNodeChannelInfo {
  return (info as VirtualNodeChannelInfo).virtual !== undefined;
}

if (isVirtual(chInfo)) {
   chInfo.value = chInfo.method(time);
}
```

anyway, not necessary, just a  thought.

Reply:
(note, in this specific example obviously it's actually more lines of code, which is a fine reason to keep things simpler instead.)

Reply:
I also thought about just having the `method` and simply checking on it's existence, but that felt a bit confusing to be passing around in the node data and to the lambda and less intuitive to use as a check for if the sensor is virtual.

Reply:
On that point, for the next phase that I'm working on, I need to set and store this notion of if something is or is not "virtual" in the node data (set from the sensor select dropdown) and later pass it to the lambda when the program is run.  I'm less inclined to try to store and pass objects through the node data when a boolean will suffice.  Granted if we can pass an object with a function to the lambda it might mean we don't need to maintain the virtual method for data generation in two places.  I would vote to leave this for now, and see if your refactoring makes sense once I complete the work to store the virtual state in the node data and pass it to the lambda.